crates/cow-amm/src/lib.rs

```diff
@@ -0,0 +1,34 @@
+mod event_updater;
+mod implementations;
+mod indexer;
+
+use {
+    crate::indexer::CowAmmRegistry,
+    ethcontract::{common::DeploymentInformation, Address},
+};
+pub use {event_updater::EventUpdater, implementations::safe_based::*, indexer::Indexer};
```

Comment:
I think the `Indexer` and `EventUpdater` should not be both exported.
My intuition would be that we export a type `Registry` which builds an `Indexer` under the hood and whatever else it needs.
And then we should be able to get all that we need from public functions on that `Registry` type.